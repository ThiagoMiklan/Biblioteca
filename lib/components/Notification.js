"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _Button = _interopRequireDefault(require("./Button.js"));

var _Content = _interopRequireDefault(require("./Content"));

var _type_validations = require("../tools/type_validations.js");

var _props_obj;

var props_obj = (_props_obj = {
  delete: 'boolean',
  title: "string",
  onClickDelete: "function",
  definition: "string"
}, (0, _defineProperty2.default)(_props_obj, "title", "string"), (0, _defineProperty2.default)(_props_obj, "subtitle", "string"), _props_obj);

var Notification = function Notification(props) {
  (0, _type_validations.validate)(props, props_obj, "Notification");
  var definition = props.definition == undefined ? "notification" : "notification " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.delete == true && assembleDelete(props.onClickDelete), props.title == undefined && props.subtitle == undefined ? props.children : assembleTitleAndSubtitle(props));
};

function assembleTitleAndSubtitle(props) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", {
    className: "title"
  }, props.title), /*#__PURE__*/React.createElement("p", {
    className: "subtitle"
  }, props.subtitle), /*#__PURE__*/React.createElement(_Content.default, null, props.children));
}

function assembleDelete(onClickDelete) {
  return /*#__PURE__*/React.createElement(_Button.default, {
    delete: true,
    onClick: onClickDelete
  });
}

var _default = Notification;
exports.default = _default;