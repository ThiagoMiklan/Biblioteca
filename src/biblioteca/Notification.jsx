import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import Button from './Button.jsx';


const Notification = (props) =>{
    var definition = "notification " + classnames(getClassName(props.definition,"Notification"));
    return(
            <div className={definition}>
                {props.delete == true && assembleDelete(props.onClickDelete)}
                {props.children}
            </div>
    );
    
}

function assembleDelete(onClickDelete){
    return <Button onClick={onClickDelete} delete={true}></Button>
}

Notification.propTypes = {
    definition: PropTypes.string,
    delete : PropTypes.bool,
    onClickDelete : PropTypes.func
}

export default Notification;