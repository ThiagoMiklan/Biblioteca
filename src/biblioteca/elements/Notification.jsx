import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import Button from './Button.jsx';


class Notification extends React.Component{

    render(){
       
            var definition = "notification " + classnames(getClassName(this.props.definition,"Notification"));
            var value_ = this.props.value;
        return(
            <div className={definition}>
                {addButtonDelete(this.props)}
                {this.props.children}
            </div>
        );
    }
}
// consertar isso, button sem enviar onClickNÃ£o aceita
function onClick(){


}
function addButtonDelete(props){
    var button_del_code = "";
    if(props.delete == true){
        button_del_code = <Button definition="delete" onClick ={onClick}/>
    }

    return button_del_code;
}

Notification.propTypes = {
    definition: PropTypes.string,
    delete : PropTypes.bool
}

export default Notification;