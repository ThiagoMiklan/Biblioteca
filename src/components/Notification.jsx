// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Button from './Button.jsx';
import Content from './Content';
import { validate } from '../tools/type_validations.js';

type Props = {
    delete?: bool,
    title?: string,
    onClickDelete?: () => void,
    definition: string,
    children: React.Node,
    title?: string,
    subtitle?: string
}
const props_obj = {
    delete: 'boolean',
    title: "string",
    onClickDelete: "function",
    definition: "string",
    title: "string",
    subtitle: "string"
}
const Notification = (props: Props) => {
    validate(props, props_obj);
    var definition = (props.definition == undefined) ? "notification" : "notification " + props.definition;
    return (
        <div className={definition}>
            {props.delete == true && assembleDelete(props.onClickDelete)}
            {props.title == undefined && props.subtitle == undefined ? props.children :
                assembleTitleAndSubtitle(props)
            }
        </div>
    )

}

function assembleTitleAndSubtitle(props) {
    return <>
        {<p className="title">{props.title}</p>}
        {<p className="subtitle">{props.subtitle}</p>}
        {<Content>{props.children}</Content>}
    </>
}

function assembleDelete(onClickDelete) {
    return <Button delete={true} onClick={onClickDelete}></Button>
}

Notification.propTypes = {
    definition: PropTypes.string,
    delete: PropTypes.bool,
    onClickDelete: PropTypes.func,
    title: PropTypes.string
}

export default Notification;