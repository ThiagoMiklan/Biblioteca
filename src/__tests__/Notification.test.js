import React from 'react';
import Notification from '../biblioteca/elements/Notification';
import renderer from 'react-test-renderer';

//  ALERTA: NOTIFICATION IS-LIGHT NÃO FUNCIONANDO, NEM COM O CÓDIGO DA DOCUMENTAÇÃO BULMA

function onClick(){
    alert("testando delete");
}
test('test colors light notification',()=>{
    const notifications_light =  renderer.create(
        <>
        <Notification definition="primary light" delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="link light" delete={true} onClickDelete={onClick}>
    Lorem ipsum dolor sit amet, consectetur
    adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
    consectetur adipiscing elit
  </Notification>

  <Notification definition="info light" delete={true} onClickDelete={onClick}>
    Lorem ipsum dolor sit amet, consectetur
    adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
    consectetur adipiscing elit
  </Notification>

  <Notification definition="success light" delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="warning light" delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="danger light" delete={true} onClickDelete={onClick}>
        Lorem ipsum dolor sit amet, consectetur
        adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
        consectetur adipiscing elit
  </Notification>
     </>
    ).toJSON();

    expect(notifications_light).toMatchSnapshot();
});


test('test colors notification',()=>{
    const notifications =  renderer.create(
     <>
       <Notification delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="primary" delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="link" delete={true} onClickDelete={onClick}>
    Lorem ipsum dolor sit amet, consectetur
    adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
    consectetur adipiscing elit
  </Notification>

  <Notification definition="info" delete={true} onClickDelete={onClick}>
    Lorem ipsum dolor sit amet, consectetur
    adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
    consectetur adipiscing elit
  </Notification>

  <Notification definition="success" delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="warning" delete={true} onClickDelete={onClick}>
      Lorem ipsum dolor sit amet, consectetur
      adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
      consectetur adipiscing elit
  </Notification>

  <Notification definition="danger" delete={true} onClickDelete={onClick}>
        Lorem ipsum dolor sit amet, consectetur
        adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a>felis venenatis</a> efficitur. Sit amet,
        consectetur adipiscing elit
  </Notification>
 
     </>
    ).toJSON();

    expect(notifications).toMatchSnapshot();
});