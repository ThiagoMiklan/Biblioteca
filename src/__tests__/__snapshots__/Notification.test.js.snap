// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test colors light notification 1`] = `
Array [
  <div
    className="notification is-primary is-light"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-link is-light"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-info is-light"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-success is-light"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-warning is-light"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-danger is-light"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
]
`;

exports[`test colors notification 1`] = `
Array [
  <div
    className="notification "
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-primary"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-link"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-info"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-success"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-warning"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
  <div
    className="notification is-danger"
  >
    <button
      className="delete"
      disabled={false}
      onClick={[Function]}
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. 
    <strong>
      Pellentesque risus mi
    </strong>
    , tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum 
    <a>
      felis venenatis
    </a>
     efficitur. Sit amet, consectetur adipiscing elit
  </div>,
]
`;
